### DISCUSSION TITLE
tfly problems with changing the world

### DISCUSSION THREAD

**Discussion by LindAn79** [Zrips/CMI#4074] [Q&A]
**Question: **
sry for the bad english but i try that with the google translated o.O.
I have a problem because of the tfly and the change of worlds!
i give the tfly rights by vote for a certain period of time and i have fly on 2 of 4 worlds. the problem is now when the players switch between the worlds they no longer have a tfly when they switch from a nofly world to a world where fly is allowed. some of the things that help is relogging, but that can't be the solution ?! can I somehow adjust that it works smoothly?

**Comment by LindAn79** [comment]
doesn't have a solution for me?

**Comment by CloudeLecaw** [comment]
you could try disable plugins that might change this, multiverse can change flymode on worldchange or playerinventorie plugins by world can do this.

if its cmi that is causing this, then you can ask for remainng tfly and set tfly on worldchange with remaining time aggain (what will overwrite the old remain time with the new value.
for the command use the playerWorldChange event of eventCommands.yml
also use it mulpie times for each world where u allow fly with ifinworld:[WorldName]! before the command
use one command for each world where u allow tfly
use -s argument so players dont see that their old time get overwritten with a new time

for checking remaining time you can use  %cmi_user_tfly%
so command would be for Flyworld
`check:%cmi_user_tfly%!=0! ifinworld:Flyworld! tfly [playerName] %cmi_user_tfly% -s`
dont forget to use this for every world where tfly is allowed and use thee correct wolrdnames instead of Flyworld ;)

**Comment by Zrips** [comment]
@LindAn79 
You need to test it out without any plugin and CMI only to make sure that tfly gets disabled by another plugin or its an issue with CMI itself. As @CloudeLecaw pointed out, plugins like multiverse could be changing fly mode while changing world, some specific configurations of CMI could do that too, so we would need to know if this issue is from CMI or another plugin to get to some solution.

**Comment by LindAn79** [comment]
okay, I actually have multiverse on it! I'll test it without a multiverse

**Comment by LindAn79** [comment]
I was able to solve the problem by setting the fly rights via cmi and no longer in mv
